db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/DMLModStatementNode.java

```diff
@@ -1557,8 +1557,6 @@ public void optimizeStatement() throws StandardException
 			if (colBitSet != null)
 			{
 				for (int i = 0; i < cols.length; i++) {
-                    // Do not Add Primary Keys, we have them in the rowKey (BaseTable) or in the RowLocation (IndexTable)
-                    if (!ArrayUtils.contains(primaryKeyColumns,cols[i]))
```

Comment:
This change basically reverts https://splicemachine.atlassian.net/browse/DB-3905 , right? Wouldn't that hurt us on TPCC?
Pinging @jleach4  since he made that change.